python/tests/catalog/test_base.py

```diff
@@ -0,0 +1,418 @@
+        removals = {} if not removals else removals
+        updates = [] if not updates else updates
+        if namespace in self.__namespaces:
+            [self.__namespaces[namespace].pop(key) for key in removals]
+            self.__namespaces[namespace].update(updates)
+        else:
+            raise NoSuchNamespaceError(f"Namespace {namespace} not found in the catalog")
+
+    @staticmethod
+    def name_from(identifier: Union[str, Identifier]) -> str:
+        return InMemoryCatalog.identifier_to_tuple(identifier)[-1]
+
+    @staticmethod
+    def namespace_from(identifier: Union[str, Identifier]) -> Identifier:
+        return InMemoryCatalog.identifier_to_tuple(identifier)[:-1]
+
+    @staticmethod
+    def identifier_to_tuple(identifier: Union[str, Identifier]) -> Identifier:
+        return identifier if isinstance(identifier, tuple) else tuple(str.split(identifier, "."))
+
+
+TEST_TABLE_IDENTIFIER = ("com", "organization", "department", "my_table")
+TEST_TABLE_NAMESPACE = ("com", "organization", "department")
+TEST_TABLE_NAME = "my_table"
+TEST_TABLE_SCHEMA = Schema(schema_id=1)
+TEST_TABLE_LOCATION = "protocol://some/location"
+TEST_TABLE_PARTITION_SPEC = PartitionSpec()
+TEST_TABLE_PROPERTIES = {"key1": "value1", "key2": "value2"}
+
+
+def given_catalog_has_a_table(catalog: InMemoryCatalog) -> Table:
+    return catalog.create_table(
+        identifier=TEST_TABLE_IDENTIFIER,
+        schema=TEST_TABLE_SCHEMA,
+        location=TEST_TABLE_LOCATION,
+        partition_spec=TEST_TABLE_PARTITION_SPEC,
+        properties=TEST_TABLE_PROPERTIES,
+    )
+
+
+def test_namespace_from_tuple():
+    # Given
+    identifier = ("com", "organization", "department", "my_table")
+    # When
+    namespace_from = InMemoryCatalog.namespace_from(identifier)
+    # Then
+    assert namespace_from == ("com", "organization", "department")
+
+
+def test_namespace_from_str():
+    # Given
+    identifier = "com.organization.department.my_table"
+    # When
+    namespace_from = InMemoryCatalog.namespace_from(identifier)
+    # Then
+    assert namespace_from == ("com", "organization", "department")
+
+
+def test_name_from_tuple():
+    # Given
+    identifier = ("com", "organization", "department", "my_table")
+    # When
+    name_from = InMemoryCatalog.name_from(identifier)
+    # Then
+    assert name_from == "my_table"
+
+
+def test_name_from_str():
+    # Given
+    identifier = "com.organization.department.my_table"
+    # When
+    name_from = InMemoryCatalog.name_from(identifier)
+    # Then
+    assert name_from == "my_table"
+
+
+def test_create_table(catalog: InMemoryCatalog):
+    table = catalog.create_table(
+        identifier=TEST_TABLE_IDENTIFIER,
+        schema=TEST_TABLE_SCHEMA,
+        location=TEST_TABLE_LOCATION,
+        partition_spec=TEST_TABLE_PARTITION_SPEC,
+        properties=TEST_TABLE_PROPERTIES,
+    )
+    assert catalog.load_table(TEST_TABLE_IDENTIFIER) == table
```

Comment:
When creating a table, I think this should validate that the table looks as expected by validating the schema, partition spec, location, and properties.

Reply:
Adding them in #3227 is okay. FYI, you may want to look at the tests we use to [validate catalogs](https://github.com/apache/iceberg/blob/master/core/src/test/java/org/apache/iceberg/catalog/CatalogTests.java) in the Java implementation. Many of the cases are the same to make sure you're getting all of the checks.